packages/block-editor/src/components/inserter/menu.js

```diff
@@ -283,7 +284,22 @@ function InserterMenu(
 		showMediaPanel,
 	] );
 
+	const { resetZoomLevel, setZoomLevel, __unstableSetEditorMode } = unlock(
+		useDispatch( blockEditorStore )
+	);
+
+	const handleZoomOut = ( newSelectedTab ) => {
+		if ( isZoomOutMode && newSelectedTab === 'blocks' ) {
+			resetZoomLevel();
+			__unstableSetEditorMode( 'edit' );
+		} else if ( ! isZoomOutMode && newSelectedTab === 'patterns' ) {
+			setZoomLevel( 50 );
+			__unstableSetEditorMode( 'zoom-out' );
+		}
+	};
```

Comment:
We have a [`useZoomOut` hook](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/use-zoom-out.js#L18) that can initialize and exit zoom out back to wherever you were. Pass it the condition if you want to enter zoom out `useZoomOut(true)` or not `useZoomOut(false)`.

Reply:
Is that hook useful here? I want to do specifically what the code says, using the hook implies I rely on the code _there_ to do something the hook does (including setting an effect). Plus the conditional execution we need here.

Reply:
I think it should disengage zoom-out when closing the inserter if it wasn't set by the user initially:
- Start in non-zoomed mode
- Open inserter
- Click Patterns
- Zoom out initializes
- Close inserter
- Zoom out returns to non-zoomed mode

Reply:
So we need both!

Reply:
I think letting the `useZoomOut` hook handle it, but setting the condition that you want to turn on/off zoom out as a const, like (pseudo code):

```
const shouldUseZoomOut = selectedTab === 'patterns';
useZoomOut( shouldUseZoomOut );
```

However, do we _always_ want to exit zoom out when the selected tab is blocks? Or _only if we automatically initiated zoom out_?

For example:
- Press zoom out from top bar
- Open Inserter
- Zoom out should not exit
- Click Patterns tab
- Click Blocks tab
- Zoom out should still not exit

Path we _do_ want to exit zoom out:
- Zoom out off
- Open inserter
- Click Patterns tab
- Zoom out 
- Click Blocks tab
- Exit zoom out

Reply:
I don't think the view should exit when the inserter is closed because the patterns tab remains selected if you open it again?

Reply:
I think the tab state is local to the inserter. It can be initiated to a tab when it opens, but I don't believe it remembers its previously selected tab.

Reply:
Oh - t[his is what is handling setting the tab to patterns](https://github.com/WordPress/gutenberg/blob/try/zoom-out-view-in-patterns-tab/packages/block-editor/src/components/inserter/menu.js#L75-L77) if opening the inserter while in zoom out:

Reply:
I did it in 14ba0f7687f52222861dc66e1c0efcd34916aab4 - the hook also handles the interplay of toggle and inserter. Thank you for persisting 🙇🏻 👏🏻